tests/DependencyInjection/SonataNewsExtensionTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Tests\DependencyInjection;
+
+use Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractExtensionTestCase;
+use Sonata\EasyExtendsBundle\Mapper\DoctrineCollector;
+use Sonata\MediaBundle\Model\Media;
+use Sonata\NewsBundle\DependencyInjection\SonataNewsExtension;
+use Sonata\NewsBundle\Model\Comment;
+use Sonata\NewsBundle\Model\Post;
+use Sonata\UserBundle\Model\User;
+
+class SonataNewsExtensionTest extends AbstractExtensionTestCase
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->container->setParameter('kernel.bundles', []);
+    }
+
+    /**
+     * Test if the deprecation notice is trigger when the tag (or collection) class declaration is missing.
+     * It should trigger a deprecation notice but doesn't break anything
+     * You should have 0 associations.
+     *
+     * @group legacy
+     * @expectedDeprecation The class %s is not defined or doesn't exist. This is tolerated now but will be forbidden in 4.0
+     */
+    public function testLoadWithoutTagWithoutCollection(): void
+    {
+        $this->load($this->getMinimalConfiguration());
+        $collector = DoctrineCollector::getInstance();
+
+        $this->assertEmpty($collector->getAssociations());
+    }
+
+    /**
+     * Test if the deprecation notice is trigger when the tag (or collection) class declaration is present.
+     * It shouldn't trigger a deprecation notice but doesn't break anything
+     * You should have 2 associations (Post, Comment).
+     * The Post model should have an associations with (Media, User, Collection, Tag).
+     */
+    public function testLoadWithTagWithCollection(): void
+    {
+        $minimalConfiguration = $this->getMinimalConfiguration();
+        $tagAndCollectionDeclaration = [
+                'class' => [
+                        'collection' => \stdClass::class,
+                        'tag' => \stdClass::class,
+                ],
+            ];
+        $this->load(array_merge($minimalConfiguration, $tagAndCollectionDeclaration));
+        $collector = DoctrineCollector::getInstance();
+        //assert our model have associations
+        $this->assertCount(2, $collector->getAssociations());
```

Comment:
Can you simplify this? This is not very readable

Reply:
I added getConfigurationWithTagWithCollection() which do the merge